ttnn/binary.py

```diff
@@ -0,0 +1,205 @@
+# SPDX-FileCopyrightText: © 2023-24 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+import sys
+import tt_lib as ttl
+from ttnn.core import reshape, _reshape_to_4D
+from ttnn.decorators import decorate_operation
+from typing import Union
+from ttnn.tensor import (
+    Tensor,
+    has_storage_type_of,
+    MemoryConfig,
+    DRAM_MEMORY_CONFIG,
+    DEVICE_STORAGE_TYPE,
+)
+from ttnn.core import reshape, _reshape_to_4D
+from ttnn.decorators import decorate_operation
+import torch
+import torch.nn.functional as F
+
+THIS_MODULE = sys.modules[__name__]
+
+__all__ = []
+
+
+def register_ttl_binary_function(name, ttl_binary_function, torch_function):
+    def _torch_binary(input_tensor_a: Tensor, input_tensor_b: Tensor, **_):
+        import torch
+        import ttnn
+
+        input_tensor_a = ttnn.from_device(input_tensor_a)
+        input_tensor_a = ttnn.to_layout(input_tensor_a, ttnn.ROW_MAJOR_LAYOUT)
+        input_tensor_a = ttnn.to_torch(input_tensor_a)
+
+        input_tensor_b = ttnn.from_device(input_tensor_b)
+        input_tensor_b = ttnn.to_layout(input_tensor_b, ttnn.ROW_MAJOR_LAYOUT)
+        input_tensor_b = ttnn.to_torch(input_tensor_b)
+        assert torch_function, f"Torch function not implemented for {str(ttl_binary_function)}"
+        return torch_function(input_tensor_a, input_tensor_b)
+
+    @decorate_operation(torch_function=_torch_binary, name=name)
+    def binary_function(
+        input_tensor_a: Tensor, input_tensor_b: Tensor, *, memory_config: MemoryConfig = DRAM_MEMORY_CONFIG
+    ) -> Tensor:
+        f"""{name}(input_tensor_a: Tensor, input_tensor_b: Tensor) -> Tensor
+
+        Applies {name} to :attr:`input_tensor_a` and  :attr:`input_tensor_b` element-wise.
```

Comment:
Looks like the documentation is unique per category of op.   This means that it will not be possible to easily add custom documentation that expands the definition.